blog/how-to-virtualize-webassembly-components-with-wasi-virt/index.mdx

```diff
@@ -0,0 +1,170 @@
+
+* Components that are composed together at build-time are tightly coupled and will generally be updated simultaneously, and will scale together once deployed. I
+* If this doesn't make sense for your application, and you don't require an otherwise unavailable core operation system interface, it may make sense to explore dynamically [linking components at runtime](/docs/concepts/linking-components/linking-at-runtime). 
+
+But returning to our original example: what if we need WASI Virt to make use of environment variables? Let's take a look at exactly how we can do it.
+
+## Example: Virtualized environment variables
+
+We need access to environment variables, so we're going to use WASI Virt to give us access to this sensitive interface via an encapsulating component. 
+
+Before we get started, we need local installations of WASI Virt and [wasmCloud Shell (wash)](https://wasmcloud.com/docs/installation). WASI Virt requires the nightly release channel for [Rust](https://www.rust-lang.org/tools/install):
+
+```bash
+rustup toolchain install nightly
+```
+
+Install the `wasi-virt` CLI tool with `cargo`:
+
+```bash
+cargo +nightly install --git https://github.com/bytecodealliance/wasi-virt
+```
+
+The example files are included in the wasmCloud [monorepo](https://github.com/wasmcloud/wasmcloud). Perform a "sparse" checkout to download only the directory for this example:
+
+```bash
+git clone --depth 1 --no-checkout https://github.com/wasmCloud/wasmCloud.git
+cd wasmcloud
+git sparse-checkout set ./examples/rust/composition
+git checkout
+cd examples/rust/composition
+```
+
+For this example, we only need the `pong` directory.
+
+### Virtualizing a component
+
+Change directory to the `pong` folder. In this directory is the code for a Rust component with a custom interface called `pong` that will return a string "ping" on its exported `pingpong` interface. Build the component:
+
+```shell
+cd pong
+wash build
+```
+
+Use `wash inspect` to look at the [WebAssembly Interface Type (WIT) interfaces](/docs/concepts/interfaces) used by this component:
+
+```shell
+wash inspect --wit ./build/pong_s.wasm
+```
+```wit
+package root:component;
+
+world root {
+  import wasi:cli/environment@0.2.0;
+  import wasi:cli/exit@0.2.0;
+  import wasi:io/error@0.2.0;
+  import wasi:io/streams@0.2.0;
+  import wasi:cli/stdin@0.2.0;
+  import wasi:cli/stdout@0.2.0;
+  import wasi:cli/stderr@0.2.0;
+  import wasi:clocks/wall-clock@0.2.0;
+  import wasi:filesystem/types@0.2.0;
+  import wasi:filesystem/preopens@0.2.0;
+  import wasi:random/random@0.2.0;
+
+  export example:pong/pingpong;
+}
+```
+The `pingpong` interface is listed as an export and `wasi:cli/environment` is among the imports. 
+
+Now we can use WASI Virt to compose the `pong_s.wasm` binary into an encapsulating component with an environment variable `PONG` set to `demo`. Our new, composed component will be named `virt.wasm`.
+
+```shell
+wasi-virt build/pong_s.wasm --allow-random -e PONG=demo -o virt.wasm
```

Comment:
```suggestion
wasi-virt build/pong_s.wasm --allow-random -e PONG=virtualized -o virt.wasm
```

It's a small thing, but I was thinking it may be nice to have a more special PONG message 😄

Reply:
(also on L115 if you do make this change)